src/Workspaces/Remote/Core/RemoteHostAssetSerialization.cs

```diff
@@ -55,14 +55,15 @@ internal static class RemoteHostAssetSerialization
 
             return;
 
-            static void WriteAsset(ObjectWriter writer, ISerializerService serializer, SolutionReplicationContext context, SolutionAsset asset, CancellationToken cancellationToken)
+            static void WriteAsset(ObjectWriter writer, ISerializerService serializer, SolutionReplicationContext context, object asset, CancellationToken cancellationToken)
             {
-                Debug.Assert(asset.Kind != WellKnownSynchronizationKind.Null, "We should not be sending null assets");
-                writer.WriteInt32((int)asset.Kind);
+                var kind = asset.GetWellKnownSynchronizationKind();
+                Contract.ThrowIfTrue(kind == WellKnownSynchronizationKind.Null);
```

Comment:
note: we have this contract virtually everywhere.  But i'm still finding and stamping out code that is resilient to null being passed along.